pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -722,11 +722,14 @@ Result ConsumerImpl::fetchSingleMessageFromBroker(Message& msg) {
     sendFlowPermitsToBroker(currentCnx, 1);
 
     while (true) {
-        incomingMessages_.pop(msg);
+        if (!incomingMessages_.pop(msg)) {
+            return ResultInterrupted;
+        }
+
         {
             // Lock needed to prevent race between connectionOpened and the check "msg.impl_->cnx_ ==
             // currentCnx.get())"
-            Lock localLock(mutex_);
+            Lock localLock1(mutex_);
```

Comment:
This change is not needed. The name of the local variable could shadow the variable outside the block.